FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,48 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case let .success((data, response)):
+				completion(RemoteFeedItemsMapper.map(data, from: response))
+			case .failure:
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+}
+
+private struct RemoteFeedItemsMapper {
+	struct Root: Decodable {
+		let items: [Item]
+
+		var feedImages: [FeedImage] {
+			items.map { $0.feedImage }
+		}
+	}
+
+	struct Item: Decodable {
+		let image_id: UUID
+		let image_desc: String?
+		let image_loc: String?
+		let image_url: URL
+
+		var feedImage: FeedImage {
+			FeedImage(id: image_id, description: image_desc, location: image_loc, url: image_url)
+		}
+	}
+
+	static func map(_ data: Data, from response: HTTPURLResponse) -> FeedLoader.Result {
+		guard response.statusCode == 200 else {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
+
+		do {
+			let root = try JSONDecoder().decode(Root.self, from: data)
+			return .success(root.feedImages)
+		} catch {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
```

Comment:
You can simplify the implementation of this function by grouping the branches returning `.failure(RemoteFeedLoader.Error.invalidData)`:

```suggestion
		guard response.statusCode == 200, 
		    let root = try? JSONDecoder().decode(Root.self, from: data) else { return .failure(RemoteFeedLoader.Error.invalidData) }

 		return .success(root.feedImages)
```

Reply:
That was one of my initial implementations but I had some Decoding errors and wanted to debug it and look into the thrown error.